### PR TITLE
Hid Site Dashboard and refactored component

### LINKED ISSUE #1303: Hide site listing, site creation behind login
### Steps to Reproduce

1. In a private browser, go to https://ocw-studio-rc.odl.mit.edu/
2. Click the "OCW Studio" icon in upper left
3. Click on any site in the listing, or try to create a site

### Expected Behavior

You shouldn't be able to do this. You haven't logged in.

### Actual Behavior

You can get to the site listing and site creation form (although site creation will fail). Visiting individual site pages will also fail.

### PR DESCRIPTION
**Pull request by Wassaf-Shahzad**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
- [x] Testing
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #1303 

#### What's this PR do?
Hides SiteDashboard in case of anonymous user 

#### How should this be manually tested?
- Open ocw-studio home page.
- Sign out if you are logged in.
-  Click ocw-studio logo.
- Verify SiteDashboard is not displayed.
- Verify Login Required is rendered. 

#### Screenshots (if appropriate)
Screenshot 2022-05-12 at 5 15 31 PM

### CODE REVIEW

File: static/js/pages/App.tsx [hunk 1]
```diff
@@ -47,10 +50,26 @@ export default function App(): JSX.Element {
           <Switch>
             <Route exact path="/" component={HomePage} />
             <Route exact path="/new-site" component={SiteCreationPage} />
-            <Route exact path="/sites" component={SitesDashboard} />
+            <Route exact path="/sites">
```
[the remark this answers is not in the document]
Comment (Wassaf-Shahzad) [reply static/js/pages/App.tsx:53]: Good Idea! Although I do have some thoughts. We should create ProtectedRoute as you say but shouldn't we inform the user in some way that he needs to login?
PS My first approach was to redirect the user but the redirect was soo seamless that it looked like the clicking the logo did nothing

[the remark this answers is not in the document]
Comment (ChristopherChudzicki) [reply static/js/pages/App.tsx:53]: >  We should create ProtectedRoute as you say but shouldn't we inform the user in some way that he needs to login?

Hmm, that's a good point. I just took a quick look at popular sites for reference... View these in a private browser if you're logged in for any of them 
1. https://www.linkedin.com/ ... has a logo at top which redirects to login
2. https://github.com/ ... has a log at top which redirects to login 
3. https://github.com/ ... has an extremely blatant "login" page with no links / options

The fact that (1) and (2) redirect to login without an alert makes me feel a better about us doing that. [Facebook's approach seems best to me, but would be way too big a change.]

IMO, redirecting to login with a <ProtectedRoute /> component seems more reusable + good separation of concerns, and the "too seamless redirect" issue is only an issue from the homepage. So I would go with the seamless login redirect.

[the remark this answers is not in the document]
Comment (Wassaf-Shahzad) [reply static/js/pages/App.tsx:53]: Ok thank you

**Review by ChristopherChudzicki — CHANGES_REQUESTED** [review]
👍 Looks good other than than the outdated comment

**Pull request merged**